src/figuresClasses.ts

```diff
@@ -1,19 +1,67 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color:Color,
+    public a:number,
+    public b:number,
+    public c:number,
+  ) {
+    if (a + b <= c
+      || b + c <= a
+      || a + c <= b
+      || b <= 0
+      || a <= 0
+      || c <= 0) {
+      throw new Error('Invalid sides');
```

Comment:
You need to say what exactly is invalid, not just throw error.
How should the user know where he did a mistake?